internal/remoteconfig/remoteconfig_test.go

```diff
@@ -377,3 +379,77 @@ func TestNewUpdateRequest(t *testing.T) {
 	require.Equal(t, "app-version", req.Client.ClientTracer.AppVersion)
 	require.True(t, req.Client.IsTracer)
 }
+
+// TestAsync starts many goroutines that use the exported client API to make sure no deadlocks occur
+func TestAsync(t *testing.T) {
+	require.NoError(t, Start(DefaultClientConfig()))
+	defer Stop()
+
+	for i := 10; i <= 10000; i *= 10 {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
```

Comment:
What's the rationale for testing w/ 10, 100, 1000, 10000? Could we not stick to just the 10k version?

Reply:
Small vs high load. You're right though, high load is enough.
Done in 59d23e4c3785